.goreleaser.yml

```diff
@@ -4,186 +4,65 @@ project_name: pulumi
-  mod_timestamp: '{{ .CommitTimestamp }}'
-- id: pulumi-language-java-unix
+- <<: *pulumibin
+  id: pulumi-language-java
   binary: pulumi-language-java
-  builder: prebuilt
-  goos: ['linux', 'darwin']
-  goarch: ['amd64', 'arm64']
-  goamd64: ['v1']
   prebuilt:
-    path: goreleaser-lang/java/{{ .Os }}-{{ replace .Arch "amd64" "x64" }}/pulumi-language-java
-  mod_timestamp: '{{ .CommitTimestamp }}'
-- id: pulumi-language-java-windows
-  binary: pulumi-language-java
-  builder: prebuilt
-  goos: ['windows']
-  goarch: ['amd64']
-  goamd64: ['v1']
-  prebuilt:
-    path: goreleaser-lang/java/{{ .Os }}-{{ replace .Arch "amd64" "x64" }}/pulumi-language-java.exe
-  mod_timestamp: '{{ .CommitTimestamp }}'
-- id: pulumi-language-yaml-unix
-  binary: pulumi-language-yaml
-  builder: prebuilt
-  goos: ['linux', 'darwin']
-  goarch: ['amd64', 'arm64']
-  goamd64: ['v1']
-  prebuilt:
-    path: goreleaser-lang/yaml/{{ .Os }}-{{ replace .Arch "amd64" "x64" }}/pulumi-language-yaml
-  mod_timestamp: '{{ .CommitTimestamp }}'
-- id: pulumi-language-yaml-windows
+    path: goreleaser-lang/java/{{ .Os }}-{{ replace .Arch "amd64" "x64" }}/{{ .Name }}{{ .Ext }}
+- <<: *pulumibin
+  id: pulumi-language-yaml
   binary: pulumi-language-yaml
-  builder: prebuilt
-  goos: ['windows']
-  goarch: ['amd64']
-  goamd64: ['v1']
   prebuilt:
-    path: goreleaser-lang/yaml/{{ .Os }}-{{ replace .Arch "amd64" "x64" }}/pulumi-language-yaml.exe
-  mod_timestamp: '{{ .CommitTimestamp }}'
+    path: goreleaser-lang/yaml/{{ .Os }}-{{ replace .Arch "amd64" "x64" }}/{{ .Name }}{{ .Ext }}
 
 archives:
-- id: pulumi-unix
-  wrap_in_directory: pulumi
-  builds:
-    - pulumi-language-dotnet-unix
-    - pulumi-language-go-unix
-    - pulumi-language-python-unix
-    - pulumi-language-nodejs-unix
-    - pulumi-language-java-unix
-    - pulumi-language-yaml-unix
-    - pulumi-unix
-  replacements:
-    amd64: x64
-  files:
-    - pulumi-resource-pulumi-nodejs
-    - pulumi-resource-pulumi-python
-    - pulumi-analyzer-policy
-    - pulumi-analyzer-policy-python
-    - pulumi-language-python-exec
-  name_template: "{{ .ProjectName }}-{{ .Tag }}-{{ .Os }}-{{ .Arch }}"
-- id: pulumi-windows
-  wrap_in_directory: pulumi/bin
-  builds:
-    - pulumi-language-dotnet-windows
-    - pulumi-language-go-windows
-    - pulumi-language-python-windows
-    - pulumi-language-nodejs-windows
-    - pulumi-language-java-windows
-    - pulumi-language-yaml-windows
-    - pulumi-windows
+- wrap_in_directory: pulumi
+  format_overrides:
+    - goos: windows
+      format: zip
   replacements:
     amd64: x64
-  format: zip
   files:
-    - pulumi-resource-pulumi-nodejs.cmd
-    - pulumi-resource-pulumi-python.cmd
-    - pulumi-python3-shim.cmd
-    - pulumi-python-shim.cmd
-    - pulumi-analyzer-policy.cmd
-    - pulumi-analyzer-policy-python.cmd
-    - pulumi-language-python-exec
+    - src: bin/{{ .Os }}/*
```

Comment:
Curious, was it not less error prone to specify all these scripts by name? Are we now sure they are actually included in the release build? If something goes per-shaped will they accidentally slide out of the release? Or that was the case before anyway?

Reply:
We use separate names for Windows vs Linux and MacOS.

This replaces using separate archive specifications for each, and instead pushes the prep work into a shared script. Allows us to name these files in one place (the script) and not in two places (the script **and** GoReleaser config).